examples/thermostats/thermostats.py

```diff
@@ -0,0 +1,900 @@
+    maximum_lag=600,
+    length_zeropadding=2000,
+    spectral_windowing="cosine-blackman",
+    timestep=1,
+    time_units="femtosecond",
+    skip=100,
+)
+
+# and plot
+fig, ax = plt.subplots(1, 1, figsize=(4, 3), constrained_layout=True)
+ax.fill_between(
+    acf_ref[:1200, 0] * ha2cm1,
+    (acf_ref[:1200, 1] - acf_ref[:1200, 2]) * 1e5,
+    (acf_ref[:1200, 1] + acf_ref[:1200, 2]) * 1e5,
+    color="gray",
+    label="reference",
+)
+ax.loglog(
+    acf_higamma[3][:1200] * ha2cm1,
+    acf_higamma[4][:1200] * 1e5,
+    "b-",
+    label=r"Langevin, $\tau=10$fs",
+)
+ax.set_xlabel(r"$\omega$ / cm$^{-1}$")
+ax.set_ylabel(r"$\hat{c}_{vv}$ / arb. units")
+ax.legend()
+plt.show()
+
+# %%
+# Global thermostats: stochastic velocity rescaling
+# -------------------------------------------------
+#
+# An alternative approach to sample the canonical Boltzmann
+# distribution while introducing fewer disturbances to the system
+# dynamics is to use a *global* thermostat, i.e. a scheme that
+# targets the *total* kinetic energy of the system, rather than that
+# of individual degrees of freedom.
+# We recommend the "stochastic velocity rescaling" thermostat
+# `(Bussi, Donadio, Parrinello, JCP (2007)) <https://doi.org/10.1063/1.2408420>`_
+# that acts by rescaling the total momentum vector, adding a
+# suitably distributed random noise term.
+
+
+# %%
+# Setting up a thermostat in ``i-PI``
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# Stochastic velocity rescaling is implemented in ``i-PI``
+# can be selected by setting ``mode="svr"``,  and has a
+# ``tau`` parameter that corresponds to the time scale of the
+# coupling.
+
+xmlroot = ET.parse("data/input_svr.xml").getroot()
+print("        " + ET.tostring(xmlroot.find(".//thermostat"), encoding="unicode"))
+
+# %%
+# We run a simulation with the usual set up ...
+
+ipi_process = None
+if not os.path.exists("simulation_svr.out"):
+    ipi_process = subprocess.Popen(["i-pi", "data/input_svr.xml"])
+    time.sleep(4)  # wait for i-PI to start
+    lmp_process = [subprocess.Popen(["lmp", "-in", "data/in.lmp"]) for i in range(1)]
+
+# %%
+# ... and wait for it to finish.
+
+if ipi_process is not None:
+    ipi_process.wait()
+    lmp_process[0].wait()
+
+# %%
+# Analysis of the trajectory
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# The kinetic temperature of the  trajectory equilibrates very
+# rapidly to the target value. However, it takes a bit longer
+# (approximately 0.5 ps) to reach equipartition between O and H
+# atoms. This is an important shortcoming of global thermostats:
+# since they only target the total kinetic energy, they must rely
+# on internal energy redistribution to reach equilibrium between
+# different degrees of freedom.
+# Liquid water is a very ergodic system, in which all degrees of
+# freedom are strongly coupled, so this is not a major issue. However
+# care must be taken when modeling a quasi-harmonic crystal, or a
+# molecular system in which the coupling between molecules is weaker.
```

Comment:
@ceriottm do you have a concrete example of this, a system for which this thermostat does not work well.

Reply:
I thought that the slow equipartition between O and H was a good enough example. 
Adding a harmonic crystal would lengthen a lot an already quite long tutorial. Do you think what we have is not enough? maybe I can distort the initial configuration to make this effect more prominent?